src/app/blog/[slug]/components/SharePost.tsx

```diff
@@ -0,0 +1,43 @@
+'use client'
+
+import { clsx } from 'clsx'
+
+import { BASE_URL } from '@/constants/siteMetadata'
+
+import { Icon } from '@/components/Icon'
+import { touchTarget } from '@/components/Social'
+
+import { getSharePostSocialIcons } from '../utils/getSharePostSocialIcons'
+
+type SharePostProps = {
+  title: string
+  postTitle: string
+}
+
+export function SharePost({ title, postTitle }: SharePostProps) {
```

Comment:
I think we will also want to use this for the Digest articles.

What if we rename this to `ShareArticle` and move it to the global components folder?